### PR TITLE
Limit GPU memory used by Tensorflow

### PR DESCRIPTION
**Pull request by svola**
Limit GPU memory used by Tensorflow to a configurable fraction, e.g. 0.5

### CODE REVIEW

**Comment by jtibshirani** [comment]
Thanks @svola for the contribution, it's great to see you are trying out the repo. I would like to keep the script quite simple and general -- I think many people running the example will not be on a machine with GPUs and may even be following along on their laptop. For this reason I'm going to close the PR.

**Comment by svola** [comment]
@jtibshirani I have been running the script on my laptop and without adding the memory limitations my whole screen freezed. I haven't worked with Tensorflow before so I don't know much about the internals. But only with this setting the code was actually executable on my machine.

**Comment by jtibshirani** [comment]
Thanks for the clarification @svola, I'll re-open this PR and review it.

File: src/main.py [hunk 1]
```diff
@@ -128,7 +128,155 @@ def embed_text(text):
...
     print("Creating tensorflow session...")
-    session = tf.Session()
+    config = tf.ConfigProto()
+    config.gpu_options.per_process_gpu_memory_fraction = 0.5
+    session = tf.Session(config=config)
+    session.run(tf.global_variables_initializer())
+    session.run(tf.tables_initializer())
+    print("Done.")
+
+    client = Elasticsearch()
```
Comment (jtibshirani) [inline src/main.py]: I don't think the script needs to be duplicated here.

Comment (svola) [reply src/main.py]: I don't know why it looks like a duplication. I just added 3 lines here 

https://github.com/svola/text-embeddings/blob/limit_memory/src/main.py#L277-L279

and the config option:

https://github.com/svola/text-embeddings/blob/limit_memory/src/main.py#L268

Comment (jtibshirani) [reply src/main.py]: Hello again, the script has been duplicated in the file you linked to. Here is the first line where the script begins to repeat itself: https://github.com/svola/text-embeddings/blob/limit_memory/src/main.py#L146

Comment (svola) [reply src/main.py]: Oh so sorry. That was a copy-paste error. It's fixed now.

**Pull request merged**